pisek/jobs/jobs.py

```diff
@@ -100,11 +102,12 @@ def cancel(self) -> None:
             return  # No need to cancel
         self.state = State.canceled
         for item, _ in self.required_by:
-            item.cancel()
+            if not item.run_always:
+                item.cancel()
 
     def _check_prerequisites(self) -> None:
         """Checks if all prerequisites are finished raises error otherwise."""
-        if self.prerequisites > 0:
+        if not self.run_always and self.prerequisites > 0:
```

Comment:
I'll admit I don't know a lot about the inner workings of the pipeline, but isn't this redundant with the changes to `finish`?
`finish` has been changed to decrement `prerequisites` for `run_always` jobs, so I'd assume bypassing the `prerequisites > 0` check is unnecessary?

Reply:
For example when generator fails, testing solutions is canceled and therefore ``finish`` is not called on them. Therefore they do not substract anything from ``self.prerequisites``.

Reply:
In that case, why are the changes in `finish` needed?